en/guide/extending_csd.md

```diff
@@ -1,6 +1,121 @@
 # Extending Camera Streaming Daemon
 
-## Create a Custom Video Stream
+## Create a Custom Camera Device
+
+Camera-Streaming-Daemon has native support for auto detection of video cameras exported as Video4Linux(V4L2) devices. This covers most of regular Linux cameras, but there are some cameras that don't have V4L2 support or cases where user wants to set very specific parameters or do custom post-processing in video before exporting.
+
+For these use cases, it is possible to create a custom CameraDevice class, representing a single custom camera device.
+CameraServer class keeps a list of all created CameraComponents, each one holding a single available CameraDevice. These CameraDevice objects are used by CameraComponent class for camera control and ImageCapture and VideoStream classes to perform the image capture and streaming.
```

Comment:
I think we need a UML class diagram here showing the ownership and ordinality. I think you are saying here that a `CameraComponent` represents an actual instance of a camera to the server, and that it owns a `CameraDevice` object that it uses to access the camera - is that correct (I was wondering why the server didn't own CameraDevice object directly)

Reply:
6. Will add UML Class diagram, that will make things clear

> I think you are saying here that a `CameraComponent` represents an actual instance of a camera to the server, and that it owns a `CameraDevice` object that it uses to access the camera - is that correct

Yes, thats correct. `CameraComponent` other than talking to `CameraDevice` to control camera, also need to manage `ImageCapture`, `VideoCapture` and `VideoStream` objects. All the mavlink requests will land in `CameraComponent` , it will then forward that to the appropriate class.